### PR TITLE
feat: implementation of basic nodejs

### PR DESCRIPTION
**Pull request by MariiaBel**
1. Task: https://github.com/AlreadyBored/nodejs-assignments/blob/main/assignments/nodejs-basics/assignment.md
2.  Deadline 2024-01-30 03:00
3.  Score:  206
- [x] fs
- [x] cli
- [x] modules
- [x] hash
- [x] streams
- [x] zip
- [ ] wt
- [ ] cp

### CODE REVIEW

File: src/fs/rename.js [hunk 1]
```diff
@@ -1,5 +1,24 @@
...
+    try {
+        const WRONG_FILE_NAME = 'wrongFilename.txt',
+            PROPER_FILE_NAME = 'properFilename.md',
+            PATH = './files'
+
+        
+        await access(`${PATH}/${PROPER_FILE_NAME}`)
+            .then(() => {
+                console.log('file exist')
+                throwError()
```
Comment (ElenVasileva) [inline src/fs/rename.js:19]: this doesn't work as expected, exception isn't thrown (because it's thrown and then caught)

File: src/fs/list.js [hunk 2]
```diff
@@ -1,5 +1,18 @@
+import { readdir } from 'fs/promises'
+import { throwError } from './error.js'
+
 const list = async () => {
-    // Write your code here 
+    try {
+        const PATH = './filess'
```
Comment (ElenVasileva) [inline src/fs/list.js]: double 's'! :-)

> const PATH = './file**SS**'

File: src/cli/args.js [hunk 3]
```diff
@@ -1,5 +1,15 @@
-const parseArgs = () => {
-    // Write your code here 
+import * as readline from 'readline/promises'
+
+const parseArgs = async() => {
+    const readlineInterface = readline.createInterface({input: process.stdin, output: process.stdout})
+    const answer = await readlineInterface.question('Print arguments in format (--propName value --prop2Name value2):')
+
+    const regular = /--(\w+)\s(\w+)/g
```
Comment (ElenVasileva) [inline src/cli/args.js:7]: this regular expression doesn't work with the example from package.json "--some-arg value1"

File: src/cli/args.js
Comment (ElenVasileva) [inline src/cli/args.js:1]: Function should parse command line arguments, for example
``node src/cli/args.js --some-arg value1 --other 1337 --arg2 42``

File: src/zip/files/fileToCompress.txt
Comment (ElenVasileva) [inline src/zip/files/fileToCompress.txt:1]: The cleared file is a bit confusing during testing

File: src/fs/copy.js [hunk 4]
```diff
@@ -1,5 +1,22 @@
+import { copyFile, readdir, mkdir, access } from 'fs/promises'
+import { throwError } from './error.js'
+
 const copy = async () => {
-    // Write your code here 
+    try {
+        const PATH = './files', NEW_PATH = './files_copy'
+        await access(PATH)
```
Comment (ElenVasileva) [inline src/fs/copy.js:11]: do not need because 'readdir' throws an exception if the folder doesn't exist

File: src/cli/env.js
Comment (ElenVasileva) [inline src/cli/env.js:1]: The output is slightly different from that described in the task - each value is on a new line and there is a semicolon at the end

File: src/fs/rename.js (hunk 1 above)
Comment (MariiaBel) [reply src/fs/rename.js:19]: it is check file access and throw error only when it already access. It is in a task description.
rename.js - implement function that renames file wrongFilename.txt to properFilename with extension .md (if there's no file wrongFilename.txt or properFilename.md already exists Error with message FS operation failed must be thrown)

Comment (ElenVasileva) [reply src/fs/rename.js:19]: I meant you can delete this code

>         await access(`${PATH}/${PROPER_FILE_NAME}`)
>             .then(() => {
>                 console.log('file exist')
>                 throwError()
>             }).catch(() => {})

This code throws an exception and then catches with empty callback, so it useless

File: src/fs/list.js (hunk 2 above)
Comment (MariiaBel) [reply src/fs/list.js]: it is for checking that throw Error is working

File: src/cli/args.js (hunk 3 above)
Comment (MariiaBel) [reply src/cli/args.js:7]: there is no such case in the task description. All parameters with camelCase as this prop2Name

File: src/cli/args.js
Comment (MariiaBel) [reply src/cli/args.js:1]: You are not right in this. All params with camelCase as in task description 
args.js - implement function that parses command line arguments (given in format --propName value --prop2Name value2, you don't need to validate it) and prints them to the console in the format propName is value, prop2Name is value2

File: src/zip/files/fileToCompress.txt
Comment (MariiaBel) [reply src/zip/files/fileToCompress.txt:1]: You can add any text there. It is not affect a program

File: src/cli/env.js
Comment (MariiaBel) [reply src/cli/env.js:1]: It is a different type of coding, We do not have rules for style. My style is more clear as I think

File: src/cli/args.js
Comment (ElenVasileva) [reply src/cli/args.js:1]: Your code parses stdin instead of command line arguments

**Pull request merged**